src/http/api/views/account.follows.viewer.ts

```diff
@@ -0,0 +1,228 @@
+            return 0;
+        }
+
+        const result = await this.db('follows')
+            .where('following_id', accountId)
+            .count('*', { as: 'count' });
+
+        return Number(result[0].count);
+    }
+
+    /**
+     * Get the accounts that are following the provided account
+     *
+     * The results are ordered in reverse chronological order
+     *
+     * @param accountId id of the account
+     * @param limit limit of the query
+     * @param offset offset of the query
+     */
+    async getFollowerAccounts(
+        accountId: number,
+        limit: number,
+        offset: number,
+    ): Promise<AccountRow[]> {
+        return await this.db('follows')
+            .select([
+                'accounts.id',
+                'accounts.ap_id',
+                'accounts.name',
+                'accounts.username',
+                'accounts.avatar_url',
+            ])
+            .where('follows.following_id', accountId)
+            .innerJoin('accounts', 'accounts.id', 'follows.follower_id')
+            .limit(limit)
+            .offset(offset)
+            // order by the date created at in descending order and then by the
+            // account id in descending order to ensure the most recent follows
+            // are returned first (i.e in case multiple follows were created at
+            // the same time)
+            // @TODO: Make this configurable via the options?
+            .orderBy('follows.created_at', 'desc')
+            .orderBy('accounts.id', 'desc');
+    }
+
+    /**
+     * Get the accounts that the provided account is following
+     *
+     * The results are ordered in reverse chronological order
+     *
+     * @param accountId id of the account
+     * @param limit limit of the query
+     * @param offset offset of the query
+     */
+    async getFollowingAccounts(
+        accountId: number,
+        limit: number,
+        offset: number,
+    ): Promise<AccountRow[]> {
+        return await this.db('follows')
+            .select([
+                'accounts.id',
+                'accounts.ap_id',
+                'accounts.name',
+                'accounts.username',
+                'accounts.avatar_url',
+            ])
+            .where('follows.follower_id', accountId)
+            .innerJoin('accounts', 'accounts.id', 'follows.following_id')
+            .limit(limit)
+            .offset(offset)
+            // order by the date created at in descending order and then by the
+            // account id in descending order to ensure the most recent follows
+            // are returned first (i.e in case multiple follows were created at
+            // the same time)
+            // @TODO: Make this configurable via the options?
+            .orderBy('follows.created_at', 'desc')
+            .orderBy('accounts.id', 'desc');
+    }
+
+    /**
+     * Check if an account is following another account
+     *
+     * @param accountId id of the account to check
```

Comment:
I think if we're going to add JSDoc comments we should be adding more context

These descriptions don't give me much more than the parameter name itself - maybe we can remove this, or clarify exactly what's happening